src/httpserver/http_utils.hpp

```diff
@@ -356,15 +371,17 @@ uint16_t get_port(const struct sockaddr* sa);
  * @param prefix Prefix to identify the map
  * @param map
 **/
-void dump_header_map(std::ostream &os, const std::string &prefix, const std::map<std::string, std::string, header_comparator> &map);
+void dump_header_map(std::ostream &os, const std::string &prefix, const http::value_map &map);
+void dump_header_map(std::ostream &os, const std::string &prefix, const http::value_map_view &map);
```

Comment:
I would go one way or the other for this dump_* methods and not keep both versions. These are mostly for debug purposes.

Reply:
I would agree, but the problem is that the HttpResponse type owns its header/footer data and therefore has `value_map`s and not `value_map_view`s. So there are places where dump_header_map is used on both owning and non-owning map types.

Reply:
I wouldn't bother trying to produce a "view" version of this one as it is a method used for debugging only. Internally to the library only http_response and http_request use it (both in their internal operations) and a test for http_utils that builds the map as map of strings. I think just keeping the old ones here is fine - outside of this PR, I'll move this under "details" to make it clear that it is not something we offer as interface, anyway.

Reply:
I actually left just the view version. The test case usages are switched to view maps (trivial) which left HttpResponse as the only place data owning versions were being dumped. I added utility to convert the map to a view just for that case. I think this results in the fewest lines of code changed. The utility (to_view_map) can also just go in the details namespace when you make that change.